pkg/beacon/relay/dkg/result/publish.go

```diff
@@ -16,9 +16,15 @@ import (
 // required protocol message unmarshallers.
 // The channel needs to be fully initialized before Publish is called.
 func RegisterUnmarshallers(channel net.BroadcastChannel) {
-	channel.RegisterUnmarshaler(func() net.TaggedUnmarshaler {
+	err := channel.RegisterUnmarshaler(func() net.TaggedUnmarshaler {
 		return &DKGResultHashSignatureMessage{}
 	})
+	if err != nil {
+		logger.Errorf(
+			"could not register DKGResultHashSignatureMessage unmarshaller: [%v]",
+			err,
+		)
+	}
```

Comment:
Instead of capturing `err` from `RegisterUnmarshaler` we should go into `RegisterUnmarshaler` and change the function signature so that it does not return `error`. We changed the behavior a long time ago and this function never fails. I think we planned to do the cleanup in a followup PR but that followup PR never happened. We are now having a situation that `ReigsterUnmarshaler` always return `nil` error.